### ISSUE TITLE
Compiler bug introduced in 0.7.2.0 with events in internal partial classes

### ISSUE THREAD

**Issue by robitar** [nikhilk/scriptsharp#63]
Just ran into this after upgrading. There are a few pieces to it, so I'll just lay them out. The basic setup is I'm generating some event declarations in a partial class, and in the handwritten part I'm consuming them.

The class is internal which is the root of the problem and the workaround (making the class public avoids the problem).

This is the 'generated' part, in MyClass.g.cs

``` c#
using System;

namespace ScriptLibrary {
    internal static partial class MyClass {
        public static event Action E1;
    }
}

```

This is the 'handwritten' part, in MyClass.cs

``` c#
using System;

namespace ScriptLibrary {
    internal static partial class MyClass {
        public static event Action E2;

        public static void F() {

        }
    }
}

```

And here is the script that is generated for it (debug build, but the same problem occurs in release script too):

``` js
Type.registerNamespace('ScriptLibrary');

////////////////////////////////////////////////////////////////////////////////
// ScriptLibrary._myClass

ScriptLibrary._myClass = function ScriptLibrary__myClass() {
}
ScriptLibrary._myClass.add_e2 = function ScriptLibrary__myClass$add_e2(value) {
    ScriptLibrary._myClass.__e2 = ss.Delegate.combine(ScriptLibrary._myClass.__e2, value);
}
ScriptLibrary._myClass.remove_e2 = function ScriptLibrary__myClass$remove_e2(value) {
    ScriptLibrary._myClass.__e2 = ss.Delegate.remove(ScriptLibrary._myClass.__e2, value);
}
ScriptLibrary._myClass.f = function ScriptLibrary__myClass$f() {
}
ScriptLibrary.MyClass.add_e1 = function ScriptLibrary_MyClass$add_e1(value) {
    ScriptLibrary.MyClass.__e1 = ss.Delegate.combine(ScriptLibrary.MyClass.__e1, value);
}
ScriptLibrary.MyClass.remove_e1 = function ScriptLibrary_MyClass$remove_e1(value) {
    ScriptLibrary.MyClass.__e1 = ss.Delegate.remove(ScriptLibrary.MyClass.__e1, value);
}


ScriptLibrary._myClass.registerClass('ScriptLibrary._myClass');
ScriptLibrary._myClass.__e2 = null;
ScriptLibrary.MyClass.__e1 = null;
```

The problem is that the code generated for the event (E1 in this case) defined in one of the partial class fragments treat the class as if it were public (ScriptLibrary.MyClass.add_e1, ScriptLibrary.MyClass.__e1 = null) causing script errors.

I changed visibility of the event declarations from public to internal, but that doesn't seem to have an effect.

**Comment by nikhilk** [comment]
Fixed in 0.7.3.

**Comment by robitar** [comment]
Thanks for the fix Nikhil, I'll upgrade and check it out later this week.
